tests/integrationv2/test_fragmentation.py

```diff
@@ -0,0 +1,111 @@
+
+
+def multi_cipher_name(c):
+    return ':'.join([x.name for x in c])
+
+
+multi_cipher = [Ciphers.AES256_SHA, Ciphers.ECDHE_ECDSA_AES256_SHA]
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("multi_cipher", [multi_cipher], ids=multi_cipher_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", [Certificates.RSA_4096_SHA384, Certificates.ECDSA_384], ids=get_parameter_name)
+def test_s2n_server_low_latency(managed_process, multi_cipher, provider, protocol, certificate):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(65519)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=multi_cipher,
+        data_to_send=random_bytes,
+        insecure=True,
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.extra_flags=['--prefer-low-latency']
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+
+    # Passing the type of client and server as a parameter will
+    # allow us to use a fixture to enumerate all possibilities.
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    # The client will be one of all supported providers. We
+    # just want to make sure there was no exception and that
+    # the client exited cleanly.
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+
+    # The server is always S2N in this test, so we can examine
+    # the stdout reliably.
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert bytes("Actual protocol version: {}".format(expected_version).encode('utf-8')) in results.stdout
+        assert random_bytes in results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("multi_cipher", [multi_cipher], ids=multi_cipher_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", [Certificates.RSA_4096_SHA384, Certificates.ECDSA_384], ids=get_parameter_name)
+@pytest.mark.parametrize("frag_len", [512, 2048, 8192, 12345, 16384], ids=get_parameter_name)
+def test_s2n_server_framented_data(managed_process, multi_cipher, provider, protocol, frag_len, certificate):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(65519)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
```

Comment:
ProviderOptions should probably just have "localhost" as a sane default for "host". We only actually give a host for the endpoint tests.

Reply:
Created #2088 for this. There are certainly other defaults that can be done better, and I don't want to update every test to use different defaults as part of this PR.